pytmc/parser.py

```
@@ -324,7 +324,24 @@ def __repr__(self):
 
 
 class TopLevelProject(TwincatItem):
-    '[tsproj] '
+    '[tsproj] Containing Io, System, Motion, TopLevelPlc, etc.'
+
+    @property
+    def ams_id(self):
+        '''
+        The AMS ID of the configured target
+        '''
+        return self.attributes.get('TargetNetId', '')
```

Missing the `AMS` ID seems important enough we might want to throw an error out here?

---

I'd rather the objects themselves (TopLevelProject) not be raising exceptions in such cases, I think. Perhaps just in st.cmd generation we verify the target net ID exists, and (optionally?) blow up if unavailable. Seem reasonable?

---

Yeah a`property` that raises an error always seem slightly distasteful. Find with checking in the flow of the `st.cmd` generation, but I think I want a loud failure at some point there.

---

OK, I think I have addressed that. It's possible we could also allow specification of IP/AMS ID from the tool itself down the line 🤷‍♂

---

Right. You may not have written and built the code on the PLC you plan to deploy?

---

Mind making an issue for this feature?